### ISSUE TITLE
Prevent rake-compiler from overriding required_ruby_version?

### ISSUE THREAD

**Issue by glebm** [rake-compiler/rake-compiler#153]
I'm working on a gem that uses FFI and doesn't use libruby directly.

This means it doesn't need an `.so` file per Ruby version, and that the pre-compiled binary gems should have the same `required_ruby_version` as the original gemspec.

rake-compiler currently does this when cross-compiling:

https://github.com/rake-compiler/rake-compiler/blob/18b335a87000efe91db8997f586772150528f342/lib/rake/extensiontask.rb#L261-L264

Is there any way to override this behaviour?

**Comment by glebm** [comment]
Super-ugly workaround:

```ruby
require 'rake/extensiontask'
gem_spec = Gem::Specification.load("sassc.gemspec")

# HACK: Prevent rake-compiler from overriding required_ruby_version,
# because the shared library here is Ruby-agnostic.
# See https://github.com/rake-compiler/rake-compiler/issues/153
module FixRequiredRubyVersion
  def required_ruby_version=(*); end
end
Gem::Specification.prepend(FixRequiredRubyVersion)

Rake::ExtensionTask.new('libsass', gem_spec) do |ext|
  ...
end
```

**Comment by kou** [comment]
https://github.com/sass/sassc-ruby/commit/f1c767750a86016fcb17d08df92e060d344504da#diff-52c976fc38ed2b4e3b1192f8a8e24cffL25 is expected code for this case.

The `cross_compiling` block exists for customizing corss-compiled gemspec:
https://github.com/rake-compiler/rake-compiler/blob/18b335a87000efe91db8997f586772150528f342/lib/rake/extensiontask.rb#L282

**Comment by glebm** [comment]
Doesn't work when I override in the block. Perhaps it is invoked earlier?

**Comment by kou** [comment]
How to reproduce your problem?

**Comment by glebm** [comment]
@kou Looks like it may be possible to override it in the block (not sure why it seemed to not work for me).

Still, I'd like to add a setting for this instead, as there is another thing to check when the gem doesn't use libruby: there should only be one RUBY_CC_VERSION.

**Comment by larskanis** [comment]
`spec.required_ruby_version` can be overwritten by the registered `ext.cross_compiling` block. But there is an issue that this callback isn't called in any case, unless `ext.no_native=true` is set. This has been discussed in 
https://github.com/rake-compiler/rake-compiler/pull/171 .

Next rake-compiler-dock version will set `no_native=true` per default, so that the callback can be used for cases like this.

**Comment by larskanis** [comment]
The `required_ruby_version` can be set per cross compile block like so:

```ruby
Rake::ExtensionTask.new('my_extension', gem_spec) do |ext|
  ext.cross_compiling do |spec|
    spec.required_ruby_version = [">= 2.5", "< 3.2.dev"]
  end
end
```
The issue with not calling the cross_compiling block is solved in #171 . So can this issue be closed now?

**Comment by flavorjones** [comment]
I agree that we can probably close it, given that the workaround exists.
